### ISSUE TITLE
Document ADR format requirements

### ISSUE THREAD

**Issue by vlcinsky** [structurizr/structurizr#63]
Existing documentation of Structurizr does not specify format of ADR markdown files.

It would be very practical to know, what requirements are in place.

So far I have discovered following requirements:
- file name must be in form of `{zero-padded-integer}-{word1}-{word2...}`
- first line must have format `# {integer}. {any title}`
- any number of blank lines
- next line must start with word `Date:` and then express ISO formatted date. If this is not true, the ADR is dated as "today"
- any number of blank lines
- heading `##"status` must be present and shall be followed by one of words:
  -  "Accepted"
  - "Superseded"
  - "Deprecated"
  - "Rejected"
  - "Proposed"
   - any other word is considered as "Proposed"
- then, any number of arbitrary links to other ADR can be listed. Some words might have special meaning line "superseded" or "supersedes"

**Comment by vlcinsky** [comment]
Nice readable code, well done.

The `{DECISION_CONTENT}` contains (anywhere, not only at the top) links, which define relations to other decisions.

There are two types of links:
- typed links: with expected text
- free form links: considered simply as "references"

The typed links must be on separate line, so the line shall use the pattern: `"^{LINK_TYPE_PREFIX} {LINK_VALUE}$"`

The typed links and respective prefixes are:
- SUPERSEDED_BY: "Superseded by" or "Superceded by"
- SUPERSEDES: "Supersedes" or "Supercedes"
- AMENDED_BY: "Amended by"
- REQUIRES: "requires "
The free form "reference" links do not require any specific prefix and the link itself does not have to be the last text on the line.

Supported statuses are:
- "Proposed"
- "Superseded"
- any other? Is "Accepted" default one? Where are "Rejected" and "Deprecated"

Open issues:
- why is "requires " (line 43) not starting with upper case letter?
- "Accepted" status - is it the default one?
- Are statuses "Deprecated" and "Rejected" missing?
- make parsing configurable for other languages:
  - date prefix and date format (currently "Date" and "yyyy-MM-dd")
  - status translation (currently "Status")
  - link prefixes (see typed links above)

**Comment by simonbrowndotje** [comment]
I've just pushed some changes that will make the importer easier to extend, and it will better extract links that are created using the `adr link` command - this is now running in the latest version of Lite.

Although not documented yet, the DSL parser has a way to specify a custom importer, for example:

```
!adrs decisions com.example.MyCustomDecisionImporter
``` 

(where `MyCustomDecisionImporter` implements `com.structurizr.documentation.importer.DocumentationImporter` ... so this will provide you a way to customise anything if necessary)

**Comment by vlcinsky** [comment]
@simonbrowndotje nice generalization of ADR format and the representation got also nicely simplified:
- The histogram of decisions during time is gone - I will not miss it
- The array of possible statuses with only the active being highlighted is simplified only to the active one - looks better now
- It seems like any status is possible now - why not

The custom importer class could be replaced by extending DSL, e.g. as follows:
```
workspace "Banking System" {
    !adrs decisions
    model {}
    views {}
    terminology {
        person "Osoba"
        softwareSystem "SW systém"
        container "Modul"

    }
    adr {
        sectionPrefix {
            status "Status"
            context "Context"
        }
        date {
            prefix "Date"
            timezone "UTC"
            format "yyy-MM-dd"
        }
        status "Proposed" {
            icon "a data URI representaing ..."
            background "#dddddd"
            color "#000000"
            /* add whatever styling sounds practical */
        }
        status "Accepted" {
            icon "a data URI representaing ..."
            background "#dddddd"
            color "#000000"
        }
        status "Rejected" {
            icon "a data URI representaing ..."
            background "#dddddd"
            color "#000000"
        }
        status "Superseded" {
            alternative "Superceded"
            alternative "Superceeded"
            icon "a data URI representaing ..."
            background "#dddddd"
            color "#000000"
        }
        status "Amended" {
            icon "a data URI representaing ..."
            background "#dddddd"
            color "#000000"
        }
    }
}
```
I tried to cover whatever I found needed.

If there is a risk, that the concepts of customizing ADR will take few iterations to become stable, I would wait with introducing DSL syntax modification.

**Comment by simonbrowndotje** [comment]
Thanks. I don't plan to make the ADR import process customisable via the DSL (extending the Java code is a better way to do that), but you can change how the statuses are rendered by adding an element style (in the `views` block) as usual, for example:

```
element "Decision:Accepted" {
    background "#00ff00"
    color "#0000ff"
}
```
